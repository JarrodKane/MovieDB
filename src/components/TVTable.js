import React from "react";

//TODO: Connect up arrows for pagnation
const TVTable = props => {
  const { tvElements, tvNext, tvPrevious } = props;

  return (
    <table className="ui celled padded table  ">
      <thead className="">
        <tr className="">
          <th className="ui medium header ">Cover</th>
          <th className="ui medium header">Title</th>
          <th className="ui medium header">Year</th>
          <th className="ui medium header">Average</th>
          <th className="ui medium header">Language</th>
          <th className="ui medium header">Add/Remove</th>
        </tr>
      </thead>
      <tbody className="">{tvElements}</tbody>
      <tfoot className="">
        <tr className="">
          <th colSpan="6" className="">
            <div className="ui pagination right floated menu">
              <div className="icon item" onClick={tvPrevious}>
                <i aria-hidden="true" className="chevron left icon"></i>
              </div>
              <div className="icon item" onClick={tvNext}>
                <i aria-hidden="true" className="chevron right icon"></i>
              </div>
            </div>
          </th>
        </tr>
      </tfoot>
    </table>
  );
};

export default TVTable;
