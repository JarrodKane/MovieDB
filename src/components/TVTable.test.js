import React from "react";
import { shallow } from "enzyme";
//import renderer from "react-test-renderer";
import TVTable from "./TVTable";

/*
test("TVTable component", () => {
  const component = renderer.create(<TVTable />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
*/

describe("TVTable", () => {
  it("should render correctly with no props", () => {
    const component = shallow(<TVTable />);

    expect(component).toMatchSnapshot();
  });
});
