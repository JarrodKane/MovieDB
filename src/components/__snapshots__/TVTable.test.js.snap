// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TVTable should render correctly with no props 1`] = `
<table
  className="ui celled padded table  "
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className="ui medium header"
      >
        Cover
      </th>
      <th
        className="ui medium header"
      >
        Title
      </th>
      <th
        className="ui medium header"
      >
        Year
      </th>
      <th
        className="ui medium header"
      >
        Average
      </th>
      <th
        className="ui medium header"
      >
        Language
      </th>
      <th
        className="ui medium header"
      >
        Add/Remove
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  />
  <tfoot
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
        colSpan="6"
      >
        <div
          className="ui pagination right floated menu"
        >
          <div
            className="icon item"
          >
            <i
              aria-hidden="true"
              className="chevron left icon"
            />
          </div>
          <div
            className="icon item"
          >
            <i
              aria-hidden="true"
              className="chevron right icon"
            />
          </div>
        </div>
      </th>
    </tr>
  </tfoot>
</table>
`;
